### ISSUE TITLE
`cannot reclaim ElementId(384)` in `dioxus-core-0.5.1/src/arena.rs:57:32` when redrawing 

### ISSUE THREAD

**Issue by Niedzwiedzw** [DioxusLabs/dioxus#2343]
`panicked at /home/niedzwiedz/.cargo/registry/src/index.crates.io-6f17d22bba15001f/dioxus-core-0.5.1/src/arena.rs:57:32:
cannot reclaim ElementId(384)`

again, unable to reproduce in a separate repo despite really trying. I'm porting an app from 0.3.1 to 0.5.0 and it was all working nicely.

**Comment by Niedzwiedzw** [comment]
another clue - it happens when a pretty large subtree gets ditched - sidebar with bunch of controls gets destroyed

**Comment by dxps** [comment]
I'm encountering this particular issue just right now, while implemeting two tabs like this:

![image](https://github.com/DioxusLabs/dioxus/assets/14140226/9464a5e0-39fb-4ede-b42e-ce4b85b9ab35)

Initially is the first tab "selected" and the corresponding details (part of a component) are shown.
Clicking on the second tab works fine. Clicking back on the 1st tab the issue happens.

And here is the relevant code:

```rust
fn render_user_profile_page(username: String, ua: UserAccount) -> Element {
    //
    log::debug!(">>> [render_user_profile_page] Username: {}, UserAccount: {:?}", username, ua);
    let mut tab_to_show = use_signal(|| "primary_info".to_string());

    rsx! {
        div { class: "flex flex-col min-h-screen bg-gray-100",
            Nav { active_path: NavProps::users_section() }
            div { class: "flex flex-col min-h-screen justify-center items-center drop-shadow-2xl",
                div { class: "bg-white rounded-md p-6 mt-24 mb-8",
                    h1 { class: "text-3xl text-[#333] font-bold text-center",
                        if username == ua.username {
                            { "My Profile" }
                        } else {
                            { format!("{}'s Profile", username) }
                        }
                    }
                    // The tabs.
                    ul { class: "flex gap-4 bg-gray-100 rounded-lg my-4 p-[3.4px] w-max overflow-hidden font-sans mx-auto",
                        li {
                            class: "text-green-600 rounded-lg font-semibold text-center text-sm bg-white py-2 px-4 tracking-wide cursor-pointer",
                            onclick: move |_| tab_to_show.set("primary_info".to_string()),
                            "Primary Info"
                        }
                        li {
                            class: "text-gray-600 rounded-lg text-center text-sm hover:bg-white hover:text-lilac py-2 px-4 tracking-wide cursor-pointer",
                            onclick: move |_| tab_to_show.set("security".to_string()),
                            "Security"
                        }
                    }
                    if tab_to_show() == "primary_info".to_string() {
                        PrimaryInfo { ua }
                    } else if tab_to_show() == "security".to_string() {
                        Security { ua }
                    }
                }
            }
        }
    }
}
```

The error thrown in the _Console_ being:
```
panicked at /home/dxps/.cargo/registry/src/index.crates.io-6f17d22bba15001f/dioxus-core-0.5.1/src/arena.rs:57:32:
cannot reclaim ElementId(30)

Stack:

__wbg_get_imports/imports.wbg.__wbg_new_abda76e883ba8a5f/<@http://localhost:3000/assets/dioxus/tmc-fsrs.js:724:21
logError@http://localhost:3000/assets/dioxus/tmc-fsrs.js:259:18
__wbg_get_imports/imports.wbg.__wbg_new_abda76e883ba8a5f@http://localhost:3000/assets/dioxus/tmc-fsrs.js:723:66
tmc_fsrs-2d842ebe5f4f9d3c.wasm.console_error_panic_hook::Error::new::h1fe0e907f35a5f1f@http://localhost:3000/assets/dioxus/tmc-fsrs_bg.wasm:wasm-function[10918]:0x425204
tmc_fsrs-2d842ebe5f4f9d3c.wasm.console_error_panic_hook::hook_impl::h1fc323d1a873f16b@http://localhost:3000/assets/dioxus/tmc-fsrs_bg.wasm:wasm-function[2338]:0x2a810d
tmc_fsrs-2d842ebe5f4f9d3c.wasm.console_error_panic_hook::hook::h18af0f385abee139@http://localhost:3000/assets/dioxus/tmc-fsrs_bg.wasm:wasm-function[18303]:0x4696f2
tmc_fsrs-2d842ebe5f4f9d3c.wasm.core::ops::function::Fn::…
```

**Comment by dxps** [comment]
@ealmloff Hi Evan!

[Here is an example](https://github.com/dxps/dioxus_playground/tree/main/dioxus-issue-2343) that reproduces the issue.
Hope it helps!
Thanks!

**Comment by ealmloff** [comment]
@dxps Thanks for the reproduction. It looks like bumping dioxus to the git version of `0.5` fixes the issue:
```toml
dioxus = { git = "https://github.com/DioxusLabs/dioxus", branch = "v0.5", features = ["fullstack", "router"] }
```

**Comment by dxps** [comment]
@ealmloff I can confirm that it's working with the git version of `0.5`. Thanks for the help! :pray: 

Moving on, I faced an issue with `dioxus_sdk` not working with it. Needed for working with the local storage.
The issue was something related to how `Signal` is declared. 
I tried to figure that out, even to have [`sdk`](https://github.com/DioxusLabs/sdk) patched to use Dioxus git version `0.5`, but it also failed with the same issue, plus others

And I did [a small example](https://github.com/dxps/dioxus_playground/tree/main/dioxus-05-test-storage) with that combination and ended up using `web-sys` to have it all working.
